File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from sys import stdin
from collections import deque

N, M, V = map(int, input().split())  # N: 정점의 개수, M: 간선의 개수, V: 시작 정점
vertex = [[] for _ in range(N + 1)]

for _ in range(M):
    v1, v2 = map(int, input().split())  # v1 <-> v2
    vertex[v1].append(v2)
    vertex[v2].append(v1)

# 낮은 정점부터 방문하기 위해서 오름차순 정렬
for i in range(1, N + 1):
    vertex[i].sort()


def dfs(V):
    visited[V] = False  # 방문 처리
    print(V, end=" ")
    for nxt in vertex[V]:
        # 방문이 가능하다면 -> dfs 탐색
        if visited[nxt]:
            visited[nxt] = False  # 방문 처리
            dfs(nxt)


def bfs(V):
    deq = deque()
    deq.append(V)
    visited = [True] * (N + 1)
    visited[V] = False  # 방문처리

    while deq:
        cur_v = deq.popleft()
        print(cur_v, end=" ")
        for nxt in vertex[cur_v]:
            # 방문이 가능하다면 -> append
            if visited[nxt]:
                visited[nxt] = False  # 방문 처리
                deq.append(nxt)


visited = [True] * (N + 1)
dfs(V)
print()
bfs(V)

File: Java/백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static class Node implements Comparable<Node> {
	int idx;

	Node(int idx) {
	    this.idx = idx;
	}

	@Override
	public int compareTo(Node other) {
	    return Integer.compare(this.idx, other.idx);
	}
    }

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static int N, M, V;
    static ArrayList<ArrayList<Node>> vertex = new ArrayList<>();
    static boolean[] visited;

    public static void main(String[] args) throws IOException {

	st = new StringTokenizer(br.readLine());
	N = Integer.parseInt(st.nextToken());
	M = Integer.parseInt(st.nextToken());
	V = Integer.parseInt(st.nextToken());

	for (int i = 0; i <= N; i++)
	    vertex.add(new ArrayList<Node>());

	for (int i = 0; i < M; i++) {
	    st = new StringTokenizer(br.readLine());

	    int v1 = Integer.parseInt(st.nextToken());
	    int v2 = Integer.parseInt(st.nextToken());

	    vertex.get(v1).add(new Node(v2));
	    vertex.get(v2).add(new Node(v1));
	}

	// 각 정점에 담긴 인덱스들을 오름차순 정렬
	for (ArrayList<Node> lst : vertex)
	    Collections.sort(lst);

	visited = new boolean[N + 1];
	dfs(V);

	bw.write("\n");
	Arrays.fill(visited, false);
	bfs(V);

	bw.close();
    }

    private static void dfs(int curr) throws IOException {

	bw.write(curr + " ");
	visited[curr] = true; // 방문 처리

	for (int i = 0; i < vertex.get(curr).size(); i++) {
	    Node nxt = vertex.get(curr).get(i);
	    // 방문하지 않은 경우 -> 탐색
	    if (!visited[nxt.idx]) {
		dfs(nxt.idx);
	    }
	}
    }

    private static void bfs(int s) throws IOException {

	ArrayDeque<Integer> dq = new ArrayDeque<>();
	dq.offer(s);
	visited[s] = true; // 방문 처리

	bw.write(s + " ");
	while (!dq.isEmpty()) {

	    int curr = dq.poll();

	    for (int i = 0; i < vertex.get(curr).size(); i++) {
		Node nxt = vertex.get(curr).get(i);
		if (!visited[nxt.idx]) {
		    bw.write(nxt.idx + " ");
		    visited[nxt.idx] = true; // 방문 처리
		    dq.offer(nxt.idx);
		}
	    }

	}

    }
}